internal/winapi/zsyscall_windows.go

```diff
@@ -267,6 +268,24 @@ func GetProcessImageFileName(hProcess windows.Handle, imageFileName *uint16, nSi
 	return
 }
 
+func CreateProcessAsUser(token windows.Token, appName *uint16, commandLine *uint16, procSecurity *windows.SecurityAttributes, threadSecurity *windows.SecurityAttributes, inheritHandles bool, creationFlags uint32, env *uint16, currentDir *uint16, startupInfo *windows.StartupInfo, outProcInfo *windows.ProcessInformation) (err error) {
+	var _p0 uint32
+	if inheritHandles {
+		_p0 = 1
+	} else {
+		_p0 = 0
+	}
+	r1, _, e1 := syscall.Syscall12(procCreateProcessAsUserW.Addr(), 11, uintptr(token), uintptr(unsafe.Pointer(appName)), uintptr(unsafe.Pointer(commandLine)), uintptr(unsafe.Pointer(procSecurity)), uintptr(unsafe.Pointer(threadSecurity)), uintptr(_p0), uintptr(creationFlags), uintptr(unsafe.Pointer(env)), uintptr(unsafe.Pointer(currentDir)), uintptr(unsafe.Pointer(startupInfo)), uintptr(unsafe.Pointer(outProcInfo)), 0)
+	if r1 == 0 {
+		if e1 != 0 {
+			err = errnoErr(e1)
+		} else {
+			err = syscall.EINVAL
```

Comment:
is this done to keep the existing behavior of `errnoErr`? I see that our implementation doesn't return `EINVAL` when `e1==0`.

Reply:
I know we essentially have a fork of the mksyscall stuff from Go in the root of this repo (that we use for the same purpose, to make go bindings for syscalls) so not sure if some changes there are the reason for the different code.